aptos-move/framework/src/natives/randomness.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright © Aptos Foundation
+
+use aptos_native_interface::{
+    RawSafeNative, SafeNativeBuilder, SafeNativeContext, SafeNativeResult,
+};
+use better_any::{Tid, TidAble};
+use move_vm_runtime::native_functions::NativeFunction;
+use move_vm_types::{loaded_data::runtime_types::Type, values::Value};
+use smallvec::{smallvec, SmallVec};
+use std::collections::VecDeque;
+
+/// A txn-local counter that increments each time a random 32-byte blob is requested.
+#[derive(Tid, Default)]
+pub struct RandomnessContext {
+    txn_local_state: Vec<u8>, //   8-byte counter
```

Comment:
do we need to include some txn specific information like txn hash?

Reply:
txn hash is currently involved [here](https://github.com/aptos-labs/aptos-core/blob/9e7cf6345a3ecc02576aeffea8f3d8a6b7ce1eee/aptos-move/framework/aptos-framework/sources/randomness.move#L59).

should be enough?